RealmSwift/App.swift

```diff
@@ -81,3 +81,29 @@ public typealias Credentials = RLMCredentials
 /// application backend.
 /// This interface provides access to login and authentication.
 public typealias App = RLMApp
+
+@available(OSX 10.15, watchOS 6.0, iOS 13.0, iOSApplicationExtension 13.0, OSXApplicationExtension 10.15, tvOS 13.0, *)
+/// Use this delegate to be provided a callback once authentication with Apple has succeed or failed
+public typealias ASLoginDelegate = RLMASLoginDelegate
+
+@available(OSX 10.15, watchOS 6.0, iOS 13.0, iOSApplicationExtension 13.0, OSXApplicationExtension 10.15, tvOS 13.0, *)
+extension App {
+    /**
+     Sets the ASAuthorizationControllerDelegate to be handled by `App`
+     - Parameter controller: The ASAuthorizationController in which you want `App` to consume its delegate.
+
+     Usage:
+     ```let app = App(id: "my-app-id")
+     let appleIDProvider = ASAuthorizationAppleIDProvider()
+     let request = appleIDProvider.createRequest()
+     request.requestedScopes = [.fullName, .email]
+
+     let authorizationController = ASAuthorizationController(authorizationRequests: [request])
+     app.setASAuthorizationControllerDelegate(controller: authorizationController)
+     authorizationController.presentationContextProvider = self
+     authorizationController.performRequests()```
+    */
+    public func setASAuthorizationControllerDelegate(controller: ASAuthorizationController) {
```

Comment:
Seems this would be more idiomatic as:
`public func setASAuthorizationController(_ controller: ASAuthorizationController)`

Reply:
I think it's important to inform the user that the purpose of this call is to set the delegate, and that no other manipulation is performed on the controller.

Reply:
I see. Then the preposition should probably be `For` instead of `With`.